state/stateConsensus.go

```diff
@@ -1896,23 +1931,33 @@ func (s *State) PutNewEntries(dbheight uint32, hash interfaces.IHash, e interfac
 
 // Returns the oldest, not processed, Commit received
 func (s *State) NextCommit(hash interfaces.IHash) interfaces.IMsg {
-	c := s.Commits[hash.Fixed()]
+	c := s.Commits.Get(hash.Fixed()) //  s.Commits[hash.Fixed()]
 	return c
 }
 
-func (s *State) PutCommit(hash interfaces.IHash, msg interfaces.IMsg) {
-	e, ok1 := s.Commits[hash.Fixed()].(*messages.CommitEntryMsg)
+// IsHighestCommit will determine if the commit given has more entry credits than the current
+// commit in the commit hashmap. If there is no prior commit, this will also return true.
+func (s *State) IsHighestCommit(hash interfaces.IHash, msg interfaces.IMsg) bool {
+	e, ok1 := s.Commits.Get(hash.Fixed()).(*messages.CommitEntryMsg)
 	m, ok1b := msg.(*messages.CommitEntryMsg)
-	ec, ok2 := s.Commits[hash.Fixed()].(*messages.CommitChainMsg)
-	mc, ok2b := msg.(*messages.CommitEntryMsg)
-
-	// Keep the most entry credits.
+	ec, ok2 := s.Commits.Get(hash.Fixed()).(*messages.CommitChainMsg)
+	mc, ok2b := msg.(*messages.CommitChainMsg)
 
+	// Keep the most entry credits. If the current (e,ec) is >=, then the message is not
+	// the highest.
 	switch {
-	case ok1 && ok1b && e.CommitEntry.Credits > m.CommitEntry.Credits:
-	case ok2 && ok2b && ec.CommitChain.Credits > mc.CommitEntry.Credits:
+	case ok1 && ok1b && e.CommitEntry.Credits >= m.CommitEntry.Credits:
+	case ok2 && ok2b && ec.CommitChain.Credits >= mc.CommitChain.Credits:
 	default:
-		s.Commits[hash.Fixed()] = msg
+		// Returns true when new commit is greater than old, or if old does not exist
+		return true
```

Comment:
Won't this _always_ return true? Seems like the switch is written so that no matter what, "return true" is executed.